providers/airbyte/src/airflow/providers/airbyte/hooks/airbyte.py

```diff
@@ -26,7 +26,11 @@
 from requests import Session
 
 from airflow.exceptions import AirflowException
-from airflow.hooks.base import BaseHook
+
+try:
+    from airflow.sdk import BaseHook
+except ImportError:
+    from airflow.hooks.base import BaseHook  # type: ignore
```

Comment:
This should be stricter:


```suggestion
    from airflow.hooks.base import BaseHook  # type: ignore[no-redef]
```

Reply:
We will need both `no-redef` and `attr-defined` when using conditionally because:

- attr-defined: mypy can’t verify that BaseHook exists in airflow.hooks.base at static time.
- no-redef: assigning to the name BaseHook in both the try and except blocks

Adding both. There could be better ways to go about this, but i think i can revisit that

Reply:
Will investigate here: https://github.com/apache/airflow/issues/52677